bin/i18n/sync-in.rb

```diff
@@ -42,17 +43,66 @@ def localize_docs
   puts "Preparing docs content"
   docs_content_file = File.join(I18N_SOURCE_DIR, "docs", "programming_environments.json")
   programming_env_docs = {}
-  ProgrammingEnvironment.all.each do |env|
-    name = env.name
-    programming_env_docs[name] = {
+  # For each programming environment, name is used as key, title is used as name,
+  ProgrammingEnvironment.all.sort.each do |env| # env as short for environment
```

Comment:
I read all the code and my 2 cents:

- This is a very very nested code by nature since you are traversing the data structure to build a map with the metadata inside of each environment.
To make it easier to read and adapt, 

Could each metadata extraction be a method
  ProgrammingEnvironment.all.sort.each do |env| # env as short for environment
      UpdateEnvironmentMetadata(env, programming_env_docs)
      env.categories_for_navigation.each do |cat|
         UpdateCategoryMetadata(cat, programming_env_docs)

Reply:
Also wonder if smoething like this be easier to folow

ProgrammingEnvironment.all.sort.each do |env|
   env.categories_for_navigation.each do |cat|
      expressions =   cat[:docs]
      expressions.each do |expression| 
                  # once here, you add all the keys to the dictionary instead of level by level.